dhall/dhall.cabal

```diff
@@ -247,25 +252,20 @@ Common common
         uri-encode                                 < 1.6 ,
         vector                      >= 0.11.0.0 && < 0.13
 
+    GHC-Options: -Wall -Wcompat -Wincomplete-uni-patterns
+
     if flag(with-http)
       CPP-Options:
         -DWITH_HTTP
       if flag(use-http-client-tls)
         CPP-Options:
           -DUSE_HTTP_CLIENT_TLS
+      if flag(network-tests)
+        CPP-Options:
+          -DNETWORK_TESTS
 
-    GHC-Options: -Wall -Wcompat -Wincomplete-uni-patterns
-
-Library
-    Import: common
-    Hs-Source-Dirs: src
-    if impl(ghcjs)
-      Hs-Source-Dirs: ghcjs-src
-      Build-Depends:
-        ghcjs-base ,
-        ghcjs-xhr
```

Comment:
I believe you will need to preserve the `ghcjs-*` dependencies for the GHCJS build to succed